### PR TITLE
App shortcut for Android N

### PR DESCRIPTION
**Pull request by ghost**
Here I'm back again (if you still know me)...

Only one app shortcut for Android N to quick access to create a new post. Perhaps, if you merge it, you should also update Android standard libs.

### CODE REVIEW

**Comment by vanitasvitae** [comment]
Is this possible without updating the android libs? There is still a bug in one of those libs which prevents us from updating them :/

**Comment by gsantner** [comment]
Hi there, thanks for your PR!

Unfortunately currently not possible due to a bug in the support libraries. If we upgrade the targetSdk a update of the support libraries (to major 25) is also required. Else it may lead to some unexpected problems

File: app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -54,6 +54,7 @@
 
             <intent-filter>
                 <action android:name="android.intent.action.SEND" />
+                <action android:name="app_shortcut" />
```
Comment (gsantner) [inline app/src/main/AndroidManifest.xml]: Please use vector drawables; you can import a svg via android studio very easy with option below the normal import option

**Comment by gsantner** [comment]
Sorry, were posting at the same time. @vanitasvitae Could you check if the bug still happens on current support lib? (And especially on Copperhead, as you got it installed)

File: app/src/main/res/xml-v25/shortcuts.xml [hunk 2]
```diff
@@ -0,0 +1,22 @@
+<shortcuts xmlns:android="http://schemas.android.com/apk/res/android">
+
+    <shortcut
+        android:shortcutId="shortcut_new_post"
+        android:enabled="true"
+        android:icon="@drawable/qc_post_plus"
```
Comment (gsantner) [inline app/src/main/res/xml-v25/shortcuts.xml:6]: Please use vector drawables; you can import a svg via android studio very easy with option below the normal import option

**Comment by vanitasvitae** [comment]
@gsantner It does. You can check the bugtracker of the issue (I'll have to search for it).

Just to be sure; @scoute-dich Does this add the long click action to the launcher icon, or does it add a new launcher icon which can be placed on the home screen?

**Comment by gsantner** [comment]
Yeah, but if the bug is open theres still the chance nobody checked it for some time :D

<https://developer.android.com/guide/topics/ui/shortcuts.html>

**Comment by ghost** [comment]
It works without updating the libs. It adds the "longclick popup shortcut" feature. No additional launcher icon. I've tested it and it works fine (OnePlus One running Nougat 7.1.2).

**Comment by ghost** [comment]
Btw. What kind of bug?

**Comment by gsantner** [comment]
@scoute-dich  It may work presently, note there are a lot of errors on bottom of build.gradle if just sdk gets upped:

unbenannt

**Comment by vanitasvitae** [comment]
Thats a shame. Google should really fix that stupid bug :/

**Comment by gsantner** [comment]
Did we try the suggested workarounds of support libs bugs? I think we did, right

**Comment by vanitasvitae** [comment]
@gsantner I tried EVERYTHING 😢

**Comment by gsantner** [comment]
I think we should just keep this open, the  feature itself is a nice addition, but we currently cannot  update due to upstream issues

**Comment by vanitasvitae** [comment]
@gsantner @scoute-dich Yeah definitelly a really nice addition :)

**Comment by ghost** [comment]
So this will be merged when there's no big in the build tools anymore?

**Comment by vanitasvitae** [comment]
Correct :)

**Comment by gsantner** [comment]
The upstream bug was fixed in supportlib v26.0.0 which is now on master.
@scoute-dich if still interested in getting that into the app: would suggest to get the latest git version of dandelion and add those 2-3 changes there. Please use vector icon if possible, just use the android-studio option and select the needed icon

**Comment by vanitasvitae** [comment]
@gsantner why did you close the PR? @scoute-dich could just have updated the branch 😄

**Comment by ghost** [comment]
So what should I do?

**Comment by gsantner** [comment]
grab the latest git version and add changes

**Comment by gsantner** [comment]
Update: tried it with new supportlibs...the problem seems NOT to be fixed...

**Comment by vanitasvitae** [comment]
@gsantner you might want to take a look at the bug report and update it with details: https://issuetracker.google.com/issues/37118105

**Comment by gsantner** [comment]
@scoute-dich My words back, Sorry. It was announced to be fixed, but it wasn't fixed at all. A fix in no way, the bug still appears most of the time, which wasn't there at supportlib-24.1.0

Android >M stuff and so shortcuts have to wait until the supportlib devs get their stuff fixed. 

aeboe8o_700b

**Comment by gsantner** [comment]
@scoute-dich  I did some work on dandelion the last days, which ended in updating SDK to API27. If you want to add this feature, I'm asking you to please do a fresh clone/fork of current master and send a new PR.

**Comment by ghost** [comment]
Jeah! I don't know how much time I have the next days, but will do my best.

**Comment by ghost** [comment]
So here we go...! Must do the icons next days and then I well start the pull request.

**Comment by gsantner** [comment]
there are already icons for that stuff, just look at navdrawer xml

**Comment by ghost** [comment]
Shouldn't we respect the design guidelines (for example the white background)?

**Comment by gsantner** [comment]
as far as i can see icons are always tintend by the launcher, and the normal material icons are used. Our icons are the normal material ones too. I don't see any problem with that :D. Other than that I don't know what you mean

screenshot_20171231-125756

**Comment by ghost** [comment]
This is what happens, if I use the navdrawer icons. So I think, I'll make some matching icons in the next days.

**Comment by gsantner** [comment]
ah you mean the icon size? the icon stuff is easy, just 4x clicks in new vectoricon  in android studio

**Pull request closed without merging**